web/templates/stream_settings/stream_settings.hbs

```diff
@@ -29,7 +29,7 @@
 <div class="subscription_settings" data-stream-id="{{sub.stream_id}}">
     <div class="inner-box">
 
-        <div class="general_settings stream_section">
+        <div class="general stream_section">
```

Comment:
Does this class name need to match the name of the tab in in the toggle widget? Because these new class names are awful, unscoped names. A good choice to migrate to would be `stream_section_general`.

Reply:
I at least don't see any obvious thing that requires a class whose name is the same as the `key` for a toggle widget to be applied to the HTML for that element.

Reply:
It does need to be the same or else explicitly mapped

```
export function setup_toggler(): void {
...
        callback(_name, key) {
            $(".stream_section").hide();
            $(`.${CSS.escape(key)}`).show();
```

That last line is what shows the body of the tab, and the classname needs to match the `key`.

@timabbott  Would you prefer I set up a map here to map from e.g. "stream_section_personal" to just "personal"? Would you prefer to keep it as is? Something else?

Reply:
I made the following change to this commit in order to avoid the awful unscoped class names:

``` diff
diff --git a/web/src/stream_edit_toggler.ts b/web/src/stream_edit_toggler.ts
index eded11f0c6..2866b12173 100644
--- a/web/src/stream_edit_toggler.ts
+++ b/web/src/stream_edit_toggler.ts
@@ -16,7 +16,7 @@ export function setup_toggler(): void {
         ],
         callback(_name, key) {
             $(".stream_section").hide();
-            $(`.${CSS.escape(key)}`).show();
+            $(`[data-stream-section="${CSS.escape(key)}"]`).show();
             select_tab = key;
         },
     });
diff --git a/web/templates/stream_settings/stream_settings.hbs b/web/templates/stream_settings/stream_settings.hbs
index 4acfd435ad..58483628af 100644
--- a/web/templates/stream_settings/stream_settings.hbs
+++ b/web/templates/stream_settings/stream_settings.hbs
@@ -29,7 +29,7 @@
 <div class="subscription_settings" data-stream-id="{{sub.stream_id}}">
     <div class="inner-box">
 
-        <div class="general stream_section">
+        <div class="stream_section" data-stream-section="general">
             {{#with sub}}
             <div class="stream-settings-tip-container">
                 {{> stream_settings_tip}}
@@ -94,7 +94,7 @@
             </div>
         </div>
 
-        <div id="personal-stream-settings" class="personal stream_section">
+        <div id="personal-stream-settings" class="stream_section" data-stream-section="personal">
             <div class="subsection-header">
                 <h3 class="stream_setting_subsection_title inline-block">{{t "Personal settings" }}</h3>
                 <div id="stream_change_property_status{{sub.stream_id}}" class="stream_change_property_status alert-notification"></div>
@@ -141,7 +141,7 @@
             </div>
         </div>
 
-        <div class="subscribers stream_section">
+        <div class="stream_section" data-stream-section="subscribers">
             {{#with sub}}
             <div class="edit_subscribers_for_stream">
                 {{> stream_members}}

```